api/api.go

```diff
@@ -147,7 +147,7 @@ func (c *Config) ClientConfig(region, address string, tlsEnabled bool) *Config {
 		Address:    fmt.Sprintf("%s://%s", scheme, address),
 		Region:     region,
 		Namespace:  c.Namespace,
-		httpClient: defaultConfig.httpClient,
+		HttpClient: defaultConfig.HttpClient,
```

Comment:
ClientConfig will overwrite any custom http client with the default client. If we support setting a custom client we should propagate it to config copies.